lib/identifiers/pubmed_id.rb

```diff
@@ -1,7 +1,9 @@
 module Identifiers
   class PubmedId
     def self.extract(str)
-      str.to_s.scan(/(?<=^|[[:space:]])0*(?!0)(\d+)(?=$|[[:space:]])/).flatten
+      str = str.to_s
+      str.scan(/(?<=^|[[:space:]])0*(?!0)(\d+)(?=$|[[:space:]])/).flatten |
+        str.scan(%r{https?://(?:www\.)?ncbi.nlm.nih.gov/pubmed/(\d+)}).flatten
```

Comment:
Does this mean we wouldn't strip zero-padding from the front of a PubMed ID in a URL? Perhaps we could lift the domain prefix here into the first non-capturing group of the previous pattern? Due to restrictions in look-behinds, we might need to change it like so:

```ruby
str.scan(%r{(?:^|[[:space:]]|https?://(?:www\.)?ncbi\.nlm\.nih\.gov/pubmed/)0*(?!0)(\d+)(?=$|[[:space:]])}).flatten
```

Reply:
This might then benefit from being a long-form pattern instead, e.g.

```ruby
%r{
  (?: # Valid prefixes for PubMed IDs
    ^ | # Start of the string
    [[:space:]] | # Leading whitespace
    https?://(?:www\.)?ncbi\.nlm\.nih\.gov/pubmed/ # A PubMed URL
  )
  0* # Any number of leading zeroes
  (?!0)(\d+) # A number not beginning with a zero
  (?=$|[[:space:]]) # Either the end of the string or trailing spaces
}x
```

Reply:
Since they're different approaches with different needs I don't think they match well when we join them together. The Pubmed URL don't really cares about how the string ends: After the Pubmed ID we could have `spaces`, `/`, `?`, `#`, `$`... 
But on the other way, it's important for the zero-padded-integer extraction. We don't want to match integers ending with `?` or `#` for example.

I've added zero-leading support to the Pubmed URL but they remain separate.